presto-main/src/main/java/com/facebook/presto/sql/analyzer/RelationType.java

```diff
@@ -38,19 +38,28 @@
 @Immutable
 public class RelationType
 {
+    private final RelationId relationId;
     private final List<Field> visibleFields;
     private final List<Field> allFields;
 
     private final Map<Field, Integer> fieldIndexes;
 
-    public RelationType(Field... fields)
+    public RelationType()
```

Comment:
I added it intentionally. `RelationType` that has no fields can be anonymous or named. No-one cares.